### PR TITLE
Fix #673 Ignore dynamic class name from missing imports

### LINKED ISSUE #673: Incorrectly warning `Missing class import via use statement` for dynamic class instantiation.
- PHPMD version: 2.7.0
- PHP Version: 7.3.9
- Installation type: composer
- Operating System / Distribution & Version: Ubuntu 18.04

### Current Behavior
I get the following warning:
```
Missing class import via use statement (line '16', column '25').
```

for the following code:
```php
<?php

use Illuminate\Database\Seeder;

class GovernorPermissionsTableSeeder extends Seeder
{
    public function run()
    {
        $actionClass = config("genealabs-laravel-governor.models.action");
        $entityClass = config("genealabs-laravel-governor.models.entity");
        $ownershipClass = config("genealabs-laravel-governor.models.ownership");
        $permissionClass = config("genealabs-laravel-governor.models.permission");
        $roleClass = config("genealabs-laravel-governor.models.role");

        $superadmin = (new $roleClass)->whereName('SuperAdmin')->get()->first();
        $actions = (new $actionClass)->all();
        $ownership = (new $ownershipClass)->whereName('any')->get()->first();
        $entities = (new $entityClass)->all();

        foreach ($entities as $entity) {
            foreach ($actions as $action) {
                (new $permissionClass)->firstOrCreate([
                    "role_key" => $superadmin->name,
                    "action_key" => $action->name,
                    "ownership_key" => $ownership->name,
                    "entity_key" => $entity->name,
                ]);
            }
        }
    }
}
```

### Expected Behavior
There is no use of a specific class, as these are dynamic class instantiations. There is no way to import the class via a `use` statement.

### Steps To Reproduce:
Run PHPMD against the above code.

## Checks before submitting
* [X] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [X] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [X] I have added every step to reproduce the bug.
* [X] If possible I added relevant code examples.
* [X] This issue is about 1 bug and nothing more.
* [X] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

### PR DESCRIPTION
**Pull request by kylekatarnls**
Type: bugfix
Issue: Fix #673
Breaking change: no

### CODE REVIEW

File: src/test/php/PHPMD/AbstractTest.php [hunk 1]
```diff
@@ -187,20 +187,29 @@ protected function getFunction()
...
-                    ->getTypes()
-                    ->current()
-                    ->getMethods(),
-                pathinfo($file, PATHINFO_FILENAME)
+        $source = $this->parseSource($file);
+        $class = $source
+            ->getTypes()
+            ->current();
+
+        return $class
```
Comment (ravage84) [inline src/test/php/PHPMD/AbstractTest.php:200]: Ternary spreading 13 lines. We are not good example here.

**Review by ravage84 — DISMISSED** [review]
Just like the other PR. Piggy-bagging a PR is not :cool: 
Can't we split that in two PRs?

The actual change looks good, though. :+1:

**Comment by kylekatarnls** [comment]
For the record, they are not piggy-bagged:

The test system migration is in the first commit:
https://github.com/phpmd/phpmd/pull/782/commits/f6b6f1b68f8a44403058740e044e2e7047bc7557

And the bug fix in the second:
https://github.com/phpmd/phpmd/pull/782/commits/554a474ccbe86bcef3e1a3a4331afa46ec971bb4

If I split, I will need to create a new method for the feature if I want it to run on Travis before the migration and the 2 PRs will be in conflict. This is actually not parallel.

**Comment by kylekatarnls** [comment]
I will create a branch from the first commit so it can be reviewed separately.

**Comment by ravage84** [comment]
> For the record, they are not piggy-bagged:
> 
> The bug fix is in the first commit:
554a474
> 
> And the test system migration in the second:
f6b6f1b

Piggy-bagged as in one PR. Like this, we as reviewers can't aprove one or the other, we have to approve both.

> If I split, I will need to create a new method for the feature test if I want it to run on Travis before the migration and the 2 PRs will be in conflict. This is actually not parallel.

Yes, I am aware that this means more work on your side. But there is the notion that one should not change behavior AND tests if not necessary (e.g. in [Martin Fowlers book on Refactoring](https://martinfowler.com/tags/refactoring.html). And in this case it is not necessary.

And yes, this way it is slower, too. My priority is not speed in advancing PHPMD/PDepend but to keep the functionality while (admittedly sometimes slowly) enhancing the code base.

**Comment by kylekatarnls** [comment]
AbstractTest.php changes removed (merged via #787)

**Comment by kylekatarnls** [comment]
Testing tools of MissingImport migration is available as a stand-alone PR #792

**Review by tvbeek — APPROVED** [review]

**Comment by tvbeek** [comment]
@ravage84 It needs your approval because you requested a change :)

**Review by ravage84 — COMMENTED** [review]
@tvbeek go for it, if you think it's fine

**Pull request merged**